third_party/3/pyspark/sql/functions.pyi

```diff
@@ -221,18 +221,18 @@ def variance(col: ColumnOrName) -> Column: ...
 
 class UserDefinedFunction:
     func = ...  # type: Callable[..., Any]
-    returnType = ...  # type: DataType
-    def __init__(self, func: Callable[..., Any], returnType: DataType, name: Optional[str] = ...) -> None: ...
+    returnType = ...  # type: DataTypeOrString
```

Comment:
That's not correct. [UDF's resolve `returnType`](https://github.com/apache/spark/blob/db3e746b64a3f78ce60bcfd6f372735f574da95a/python/pyspark/sql/udf.py#L110-L114), before wrapper is returned, and don't return strings.

Reply:
fixed!